arangod/Aql/ExecutionBlockImpl.cpp

```diff
@@ -185,45 +186,228 @@ std::unique_ptr<OutputAqlItemRow> ExecutionBlockImpl<Executor>::createOutputRow(
   }
 }
 
+namespace arangodb {
+namespace aql {
+
+enum class SkipVariants { SKIPROW, SKIPROWS, EXECUTOR, DEFAULT };
+
+template <enum SkipVariants>
+struct ExecuteSkipVariant {};
+
+template <>
+struct ExecuteSkipVariant<SkipVariants::SKIPROW> {
+  template <class Executor>
+  static std::pair<ExecutionState, size_t> executeSkip(Executor& executor,
+                                                       typename Executor::Fetcher& fetcher,
+                                                       size_t toSkip) {
+    ExecutionState state;
+    InputAqlItemRow input = InputAqlItemRow{CreateInvalidInputRowHint{}};
+    ;
+
+    std::tie(state, input) = fetcher.skipRow();
+    if (state == ExecutionState::WAITING) {
+      return {state, 0};
+    } else {
+      return {state, 1};
+    }
+  }
+};
+
+template <>
+struct ExecuteSkipVariant<SkipVariants::SKIPROWS> {
+  template <class Executor>
+  static std::pair<ExecutionState, size_t> executeSkip(Executor& executor,
+                                                       typename Executor::Fetcher& fetcher,
+                                                       size_t toSkip) {
+    return fetcher.skipRows(toSkip);
+  }
+};
+
+template <>
+struct ExecuteSkipVariant<SkipVariants::EXECUTOR> {
+  template <class Executor>
+  static std::pair<ExecutionState, size_t> executeSkip(Executor& executor,
+                                                       typename Executor::Fetcher& fetcher,
+                                                       size_t toSkip) {
+    return executor().skipRows(toSkip);
+  }
+};
+
+template <>
+struct ExecuteSkipVariant<SkipVariants::DEFAULT> {
+  template <class Executor>
+  static std::pair<ExecutionState, size_t> executeSkip(Executor& executor,
+                                                       typename Executor::Fetcher& fetcher,
+                                                       size_t toSkip) {
+    // this function should never be executed
+    TRI_ASSERT(false);
+  }
+};
+
 template <class Executor>
-std::pair<ExecutionState, size_t> ExecutionBlockImpl<Executor>::skipSome(size_t atMost) {
-  // TODO IMPLEMENT ME, this is a stub!
+static SkipVariants constexpr skipType() {
+  // TODO: Add subquery modification check - still missing
+  /*if (std::is_same<Executor, SubqueryExecutor>::value && isModificationNode) {
+    return SkipVariants::DEFAULT;
+  }*/
```

Comment:
@goedderz here still not clear how to identify a subquery correctly and if it is a modification node. 
`bool SubqueryNode::isModificationSubquery()`